template/FormlessTemplate.purs

```diff
@@ -0,0 +1,129 @@
+module FormlessTemplate where
+
+import Prelude
+
+import DOM.HTML.Indexed.InputType (InputType(..))
+import Data.Maybe (Maybe(..))
+import Data.Newtype (class Newtype)
+import Effect.Aff (Aff)
+import Formless as F
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Template.DataAndValidation (NAME_FIELD, _name, minLength)
+import Type.Row (type (+))
+
+type FormFieldsRow f =
+  ( NAME_FIELD f
+  + ()
+  )
+
+type FormFields = { | FormFieldsRow F.OutputType }
+
+newtype Form r f = Form (r (FormFieldsRow f))
+derive instance newtypeForm' :: Newtype (Form r f) _
+
+-- Form component types
+
+type Input = Unit
+type AddedState = ( additionalState :: Maybe Int )
+data Action
+  = DoStuff
+  | Initialize
+  | Finalize
+  | Receive Input
+
+data Query a
+  = Command a
+  | Reply (Unit -> a)
+
+type Message = FormFields
+type ChildSlots =
+  ()
+type MonadType = Aff
+type SelfSlot index = F.Slot Form Query ChildSlots Message index
+
+component :: F.Component Form Query ChildSlots Input FormFields Aff
+component = F.component mkInput $ F.defaultSpec
+  { render = render
+  , handleAction = handleAction
+  , handleEvent = handleEvent
+  , handleQuery = handleQuery
+  , initialize = Just Initialize
+  , finalize = Just Finalize
+  , receive = Just <<< Receive
+  }
```

Comment:
I think each of these (`mkInput` included) could use a one-line description of what it's for, just as a reminder. If you would like, I can add those in for you, or feel free to do so. 

Something simple like:

```purs
-- Decide what, if anything, to do when Formless events occur. If you would like to raise events as
-- messages, then use `F.raiseResult` as your `handleEvent` function.
handleEvent :: ...
```